### PR TITLE
Fixes error in validation for facebook and twitter input fields

### LINKED ISSUE #6826: Sticky invalid state for Twitter/Facebook inputs
### Issue Summary

The "invalid" state of the twitter/facebook inputs is sticky if a valid URL is entered which doesn't need the auto-rewrite

### Steps to Reproduce

1. Visit the settings page
2. Enter `https://twitter.com/testing` into the Facebook input
3. Click out of the input, you should see an error message
4. Click back into the input and change the text to `https://www.facebook.com/testing`
5. Click out of the input, you'll still see the error message even though it's a valid URL

The bug only appears if the value entered in step 4 exactly matches the URL format that we expect, if the value needs re-writing in any way then the validation passes.

This occurs for the Twitter and Facebook inputs on both the Settings and Team/User pages.

### PR DESCRIPTION
**Pull request by aileen**
closes #6826
- refactors the validation of facebook and twitter input field in `general.js` and `user.js` controller
  - Example validations for facebook:
    - `facebook.com/username` will be corrected to the full URL
    - `user` will show error `Your Page name is not a valid Facebook Page name` for `general.js` and `Your Username is not a valid Facebook Username` for `user.js` as the username in facebook has to be at least 5 characters long
    - `facebook.com/username` will be autocorrected to the valid facebook URL incl. the `username`
  - Example validations for twitter:
    - `twitter.com/user_` will be corrected to the full URL
    - `user:99` will show error`Your Username is not a valid Twitter Username`
    - `facebook.com/username` will be autocorrected to the valid twitter URL incl. the `username`
- updates both acceptance tests
- adds further validation for facebook pages in general settings and user. Submitting a url which incl. `/page/` or `/pages/` will now accept any username followed incl. further `/`.
- adds a custom transform `facebook-url-user` which will extract the username (if it's a facebook page, incl. `pages/`) to store only this in the backend
- uses the `twitter-url-user` transform now also for user

### CODE REVIEW

**Comment by AileenCGN** [comment]
@kevinansfield Could you check this? It does even more validation/autocorrection than before...

**Comment by acburdine** [comment]
@AileenCGN not to change a bunch of things here, but why is the validation performed in the controller instead of in `core/client/app/validators/settings.js` and `core/client/app/validators/user.js`? I feel like that could simplify things a bit.

**Comment by AileenCGN** [comment]
I know, @acburdine... There's a todo for that... As far as I remember, the reason why we did it like this for now was because of the 'autocorrection' (@kevinansfield?)... I will check again, if we can refactor this.

**Comment by acburdine** [comment]
afaik, it is possible to autocorrect in the validators. It's not necessarily the _best_ behavior, but it can definitely be done 😀. I moved this validations into the validators in my inline validations refactor pr, and it seems to work ok (#6647)

**Comment by kevinansfield** [comment]
@AileenCGN this is definitely something that can be moved to a validator but it's good to first get it working the way you have it now - refactoring is iterative, once it's working and tested you can try moving the logic into a validator and the tests should guide you if any of the functionality breaks or changes 😄

File: core/client/app/controllers/settings/general.js [hunk 1]
```diff
@@ -180,39 +190,53 @@ export default Controller.extend(SettingsSaveMixin, {
...
+                this.get('model.hasValidated').pushObject('twitter');
+
+                // User input is validated
+                return this.save().then(() => {
+                    this.set('model.twitter', '');
+                    run.schedule('afterRender', this, function () {
+                        this.set('model.twitter', newUrl);
+                    });
+                });
+            } else if (validator.isURL(newUrl)) {
```
Comment (kevinansfield) [inline core/client/app/controllers/settings/general.js]: This `else if` block can be removed as it does exactly the same thing as the `else` block below.

**Comment by kevinansfield** [comment]
Aside from the duplicated conditional block mentioned above I think this is 👍 for the release. I think the next steps after would be refactoring the re-writing parts into utility functions and pulling the validation logic into a validator, or at the very least de-duplicating so that there's no copy-paste of large logic blocks between the general and user controllers.

**Comment by kevinansfield** [comment]
👍 working OK for me 😄 Definitely due some refactoring but the bugs are fixed so I think this is ready to go for the release.

**Comment by ErisDS** [comment]
Closing in favour of #6848

**Pull request closed without merging**